Mathlib/Algebra/Module/Injective.lean

```diff
@@ -446,32 +442,105 @@ theorem extensionOfMax_to_submodule_eq_top (h : Module.Baer R Q) :
-  ⟨g' ∘ₗ ULift.moduleEquiv.symm.toLinearMap, by aesop⟩
+protected theorem of_injective [UnivLE.{u, v}] (inj : Module.Injective R Q) : Module.Baer R Q := by
+  intro I g
+  let eI := Shrink.linearEquiv I R
+  let eR := Shrink.linearEquiv R R
 
-protected theorem iff_injective : Module.Baer R Q ↔ Module.Injective R Q :=
-  ⟨Module.Baer.injective, Module.Baer.of_injective⟩
+  obtain ⟨g', hg'⟩ := Module.Injective.out (eR.symm.toLinearMap ∘ₗ I.subtype ∘ₗ eI.toLinearMap)
+    (eR.symm.injective.comp <| Subtype.val_injective.comp eI.injective) (g ∘ₗ eI.toLinearMap)
+
+  exact ⟨g' ∘ₗ eR.symm.toLinearMap, fun x mx ↦ by simpa using hg' (equivShrink I ⟨x, mx⟩)⟩
+
+protected theorem iff_injective [UnivLE.{u, v}] : Module.Baer R Q ↔ Module.Injective R Q :=
+  ⟨Module.Baer.injective, Module.Baer.of_injective.{u, v}⟩
 
 end Module.Baer
+
+section ULift
+
+variable {M : Type v} [AddCommGroup M] [Module R M] [UnivLE.{u, v}]
+
+lemma Module.ulift_injective_of_injective
+    (inj : Module.Injective R M) :
+    Module.Injective R (ULift.{max v' v} M) := by
+  letI : UnivLE.{u, max v' v} := UnivLE.trans.{v, u, max v' v}
+  rw [← Module.Baer.iff_injective.{u, v}] at inj
+  rw [← Module.Baer.iff_injective.{u, max v' v}]
+  intro I g
+  obtain ⟨g', hg'⟩ := inj I (ULift.moduleEquiv.toLinearMap ∘ₗ g)
+  exact ⟨ULift.moduleEquiv.symm.toLinearMap ∘ₗ g', fun r hr ↦ ULift.ext _ _ <| hg' r hr⟩
+
+lemma Module.injective_of_ulift_injective
+    (inj : Module.Injective R (ULift.{max v v'} M)) :
+    Module.Injective R M := by
+  letI : UnivLE.{u, max v' v} := UnivLE.trans.{v, u, max v' v}
+  rw [← Module.Baer.iff_injective.{u, max v v'}] at inj
+  rw [← Module.Baer.iff_injective.{u, v}]
+  intro I g
+  obtain ⟨g', hg'⟩ := inj I (ULift.moduleEquiv.symm.toLinearMap ∘ₗ g)
+  exact ⟨ULift.moduleEquiv.toLinearMap ∘ₗ g', fun r hr ↦ ULift.ext_iff _ _ |>.mp <| hg' r hr⟩
+
+variable (M)
+
+lemma Module.injective_iff_ulift_injective :
+    Module.Injective R M ↔ Module.Injective R (ULift.{max v v'} M) :=
+  ⟨Module.ulift_injective_of_injective.{u, v, v'} R,
+   Module.injective_of_ulift_injective.{u, v, v'} R⟩
+
+instance ModuleCat.ulift_injective_of_injective
+    [inj : CategoryTheory.Injective <| ModuleCat.of R M] :
+    CategoryTheory.Injective <| ModuleCat.of R (ULift.{max v v'} M) :=
+  Module.injective_object_of_injective_module
+    (inj := Module.ulift_injective_of_injective.{u, v, v'}
+      (inj := Module.injective_module_of_injective_object (inj := inj)))
+
+end ULift
+
+section lifting_property
+
+universe uR uM uP uP'
+
+variable [UnivLE.{uR, uM}]
+variable (R : Type uR) [Ring R]
+variable (M : Type uM) [AddCommGroup M] [Module R M] [inj : Module.Injective R M]
+variable (P : Type uP) [AddCommGroup P] [Module R P]
+variable (P' : Type uP') [AddCommGroup P'] [Module R P']
+
+lemma Module.Injective.extension_property
+    (f : P →ₗ[R] P') (hf : Function.Injective f)
+    (g : P →ₗ[R] M) : ∃ h : P' →ₗ[R] M, h ∘ₗ f = g := by
```

Comment:
This is golfed via Module.Baer in my branch.

Reply:
Thanks for all the golfing